pandas/_libs/tslibs/period.pyx

```diff
@@ -946,9 +947,26 @@ def periodarr_to_dt64arr(const int64_t[:] periodarr, int freq):
         int64_t[:] out
         Py_ssize_t i, l
 
-    l = len(periodarr)
+    if freq >= FR_DAY:
+        if freq == FR_NS:
+            return periodarr
+
+        if freq == FR_US:
+            dta = periodarr.view("M8[us]")
+        elif freq == FR_MS:
+            dta = periodarr.view("M8[ms]")
+        elif freq == FR_SEC:
+            dta = periodarr.view("M8[s]")
+        elif freq == FR_MIN:
+            dta = periodarr.view("M8[m]")
+        elif freq == FR_HR:
+            dta = periodarr.view("M8[h]")
+        elif freq == FR_DAY:
+            dta = periodarr.view("M8[D]")
+        return ensure_datetime64ns(dta)
 
-    out = np.empty(l, dtype='i8')
+    l = len(periodarr)
+    out = np.empty(l, dtype="i8")
```

Comment:
this could be a nogil loop right? (actually couldn't this entire function)?

Reply:
worth doing this here?

Reply:
this cant be nogil becuase it calls check_dts_bounds, which may raise